### ISSUE TITLE
[Training] Missing 'BFLOAT16' attribute in `torch._C._onnx.TensorProtoDataType` on ROCm

### ISSUE THREAD

**Issue by hubertlu-tw** [microsoft/onnxruntime#15207]
### Describe the issue

ROCm microbenchmark fails due to missing 'BFLOAT16' attribute in `torch._C._onnx.TensorProtoDataType`.
It looks that this failure was introduced by [this PR](https://github.com/microsoft/onnxruntime/pull/13972) in [this file](https://github.com/microsoft/onnxruntime/blame/main/orttraining/orttraining/python/training/ortmodule/_custom_op_symbolic_registry.py#L30).

```bash
Traceback (most recent call last):
  File "softmax.py", line 10, in <module>
    from benchmark import BenchmarkOp, add_arguments
  File "/onnxruntime/onnxruntime/python/tools/microbench/benchmark.py", line 14, in <module>
    import onnxruntime as ort
  File "/opt/conda/lib/python3.8/site-packages/onnxruntime/__init__.py", line 53, in <module>
    from onnxruntime.capi import onnxruntime_validation
  File "/opt/conda/lib/python3.8/site-packages/onnxruntime/capi/onnxruntime_validation.py", line 145, in <module>
    has_ortmodule, package_name, version, cuda_version = validate_build_package_info()
  File "/opt/conda/lib/python3.8/site-packages/onnxruntime/capi/onnxruntime_validation.py", line 140, in validate_build_package_info
    raise import_ortmodule_exception
  File "/opt/conda/lib/python3.8/site-packages/onnxruntime/capi/onnxruntime_validation.py", line 70, in validate_build_package_info
    from onnxruntime.training.ortmodule import ORTModule  # noqa
  File "/opt/conda/lib/python3.8/site-packages/onnxruntime/training/__init__.py", line 16, in <module>
    from .ortmodule import ORTModule
  File "/opt/conda/lib/python3.8/site-packages/onnxruntime/training/ortmodule/__init__.py", line 16, in <module>
    from ._fallback import ORTModuleFallbackException, ORTModuleInitException, _FallbackPolicy, wrap_exception
  File "/opt/conda/lib/python3.8/site-packages/onnxruntime/training/ortmodule/_fallback.py", line 23, in <module>
    from . import _utils
  File "/opt/conda/lib/python3.8/site-packages/onnxruntime/training/ortmodule/_utils.py", line 29, in <module>
    from ._custom_op_symbolic_registry import CustomOpSymbolicRegistry
  File "/opt/conda/lib/python3.8/site-packages/onnxruntime/training/ortmodule/_custom_op_symbolic_registry.py", line 30, in <module>
    "BFloat16": torch.onnx.TensorProtoDataType.BFLOAT16,
AttributeError: type object 'torch._C._onnx.TensorProtoDataType' has no attribute 'BFLOAT16'
```

### To reproduce

# Dockerfile
```Dockerfile
FROM rocm/pytorch:rocm5.4_ubuntu20.04_py3.8_pytorch_1.10.2

RUN pip uninstall cmake
RUN pip install cmake==3.24.1

ARG ONNXRUNTIME_REPO=https://github.com/Microsoft/onnxruntime
ARG ONNXRUNTIME_BRANCH=main
ARG ROCM_VERSION=5.4.0

WORKDIR /var/lib/jenkins

# Prepare onnxruntime repository & build onnxruntime
RUN git clone --single-branch --branch ${ONNXRUNTIME_BRANCH} --recursive ${ONNXRUNTIME_REPO} && \
    cd onnxruntime && \
    ./build.sh --enable_training --config Release --use_rocm --rocm_version=${ROCM_VERSION} \
        --rocm_home /opt/rocm --nccl_home /opt/rocm --build_wheel --skip_tests --update --build --parallel \
        --cmake_extra_defines onnxruntime_BUILD_KERNEL_EXPLORER=ON

RUN pip install /var/lib/jenkins/onnxruntime/build/Linux/Release/dist/*.whl
ENV KERNEL_EXPLORER_BUILD_DIR=/var/lib/jenkins/onnxruntime/build/Linux/Release
```

# Build docker image and reproduce the errors
```Bash
docker build -t rocm5.4_pytorch_1.10.2_kernel_explorer -f Dockerfile .
sudo docker run -it --network=host --device=/dev/kfd --device=/dev/dri --ipc=host --shm-size 16G --group-add video --cap-add=SYS_PTRACE --security-opt seccomp=unconfined rocm5.4_pytorch_1.10.2_kernel_explorer

# This works fine
cd /var/lib/jenkins/onnxruntime/onnxruntime/python/tools/kernel_explorer/kernels/
python skip_layer_norm_test.py 1 1 4096 float16

# This fails
cd /var/lib/jenkins/onnxruntime/onnxruntime/python/tools/microbenchmark
python skip_layer_norm.py
```

### ONNX Runtime Installation

Built from Source

### ONNX Runtime Version or Commit ID

56bccac35

### PyTorch Version

1.10.2

### Execution Provider

ROCm

**Comment by thiagocrepaldi** [comment]
Try with latest pytorch, which uses newer onnx which should have bfloat16
Pytorch 1.10 is too old

**Comment by mindest** [comment]
`BFLOAT16` is not included until torch v1.11.0:
https://github.com/pytorch/pytorch/blob/v1.10.2/torch/_C/_onnx.pyi
https://github.com/pytorch/pytorch/blob/v1.11.0/torch/_C/_onnx.pyi

**Comment by baijumeswani** [comment]
Closing this. Please let us know if the problem still persists.
